Enhance iOS UI, Unify Styling, and Disable Global USD Mode

fairlighteth opened the pull request:
# Summary of Changes

## Mobile & iOS Enhancements
- Auto-scroll to the "My Orders" section on mobile devices.
- Enforced 16px font-size to prevent iOS input zoom. This is for iOS devices only using a CSS only solution.

## UI & Styling Updates
- Capitalized certain labels (e.g. “Settings”).
- Refactored components (`RateInfo`, `OrderProgressBarV2`, row elements) to standardize props and remove extra logic (e.g. `withTimelineDot`).
- Fine-tuned spacing, margins, and font sizes across various components.
- Set a fixed table height in certain views.

## Feature Toggles & Visibility
- Hid search input and tabs if no orders or if the user is not connected.
- Showed search UI even for zero results.
- Removed/Commented out code paths for the Global USD Mode toggle.
- Prevented "filled" status from displaying for orders still awaiting signature (TWAP).

## Miscellaneous Fixes
- Corrected some props being passed through the DOM.
- Improved handling of different order states (final vs. signing).
- Cleaned up tooltips, references to “timeline dots,” and standardized color usage.

Overall, these changes streamline the codebase, improve mobile/iOS usability, refine UI behaviors, and disable unneeded features for a cleaner user experience.

elena-zh reviewed (commented):
Hey @fairlighteth , great, thank you!
I have a nitpick to report: is it possible to hide 'all orders' tab to a not connected user? For now, it is clickable (it does not make any sense), and does not look good in a mobile view: 
![clicable](https://github.com/user-attachments/assets/18f04e6d-e916-4895-b923-446911dd95c0)
![image](https://github.com/user-attachments/assets/45d15901-b3da-4a42-8055-6a8f5283ba7b)

elena-zh commented:
> Originally, I removed it when the user wasn’t connected. However, keeping it serves to inform users that they can expect to see their orders there—even if none are currently displayed.

@fairlighteth , thank you for explanation. At least, could you please remove 'hover' effect from it as it does not make sense? 
![image](https://github.com/user-attachments/assets/4652e79a-374b-4c79-8e2b-5fdd49b0a100)

fairlighteth commented:
> > Originally, I removed it when the user wasn’t connected. However, keeping it serves to inform users that they can expect to see their orders there—even if none are currently displayed.
> 
> @fairlighteth , thank you for explanation. At least, could you please remove 'hover' effect from it as it does not make sense? ![image](https://private-user-images.githubusercontent.com/70885163/404888613-4652e79a-374b-4c79-8e2b-5fdd49b0a100.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NzMzNzIsIm5iZiI6MTczNzQ3MzA3MiwicGF0aCI6Ii83MDg4NTE2My80MDQ4ODg2MTMtNDY1MmU3OWEtMzc0Yi00Yzc5LThlMmItNWZkZDQ5YjBhMTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDE1MjQzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyNTkyM2Q1OGFhMWU3ZmM0MzI1ZjM2ZmI5ZGZlNjhmMDA1MmJlMmQzZWRmMTU5NDg5MWQxYzBiMjA0YjA3YTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RoTH2pbJnKIZh4DZp__PaQHO25kAyObGmzhFzefD4Nw)

Addressed in this PR

shoom3301 commented on apps/cowswap-frontend/src/modules/limitOrders/containers/TradeRateDetails/index.tsx:
```diff
@@ -46,7 +45,7 @@ export function TradeRateDetails({ rateInfoParams, alwaysExpanded = false }: Tra
     return (
       <>
-        <StyledRateInfo label="Limit price" stylized={true} rateInfoParams={rateInfoParams} />
+        <StyledRateInfo label="Limit price" stylized={true} rateInfoParams={rateInfoParams} rightAlign={true} />
```
```suggestion
        <StyledRateInfo label="Limit price" stylized={true} rateInfoParams={rateInfoParams} rightAlign />
```

shoom3301 commented on apps/cowswap-frontend/src/modules/limitOrders/pure/LimitOrdersDetails/styled.tsx:
```diff
@@ -11,40 +9,20 @@ export const DetailsRow = styled.div`
 `
 
-export const StyledRateInfo = styled(RateInfo)`
+export const StyledRateInfo = styled(RateInfo).attrs({ rightAlign: true })`
```
`rightAlign` attr is not used in this component, should it be deleted?

shoom3301 commented on apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrdersTabs.tsx:
```diff
@@ -141,9 +145,10 @@ const TabButton = styled(Link)<{ active: string; isUnfillable?: boolean; isSigni
 
 export interface OrdersTabsProps {
   tabs: OrderTab[]
+  isWalletConnected?: boolean
```
Why it can be undefined? Looks like it's always boolean

fairlighteth replied:
Addressed this.

shoom3301 commented on apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx:
```diff
@@ -39,7 +39,8 @@ import { WrapFlowActionButton } from '../WrapFlowActionButton'
   const element = document.getElementById(MY_ORDERS_ID)
   if (element) {
-    element.scrollIntoView({ behavior: 'smooth' })
+    const elementTop = element.getBoundingClientRect().top + window.scrollY - 76
```
Need comments here, what 76 does mean?

fairlighteth replied:
Addressed this. This is the sticky header offset. Made more obvious by moving to const with a comment.

shoom3301 commented on apps/cowswap-frontend/src/modules/trade/pure/TradePageLayout/index.tsx:
```diff
@@ -50,7 +50,9 @@ export const PrimaryWrapper = styled.div`
 
 // Graph + orders table
-export const SecondaryWrapper = styled.div`
+export const SecondaryWrapper = styled.div.attrs(() => ({
```
```suggestion
export const SecondaryWrapper = styled.div.attrs({
  id: MY_ORDERS_ID,
})`
```
